lib/modules/manager/poetry/artifacts.ts

```diff
@@ -187,7 +187,9 @@ export async function updateArtifacts({
       docker: {},
       toolConstraints: [{ toolName: 'python', constraint }],
       preCommands: [
-        `pip install --user ${quote(`poetry${poetryVersion ?? ''}`)}`,
+        `pip install --user ${quote(
+          `poetry${poetryVersion ? `===${poetryVersion}` : ''}`
+        )}`,
```

Comment:
convert to tool constraints instead, it's now supported by underlying containerbase image

Reply:
Thank you for the tip, that's much cleaner. I've tested locally with `binarySource=install` and it works as expected.